src/jdk.incubator.code/share/classes/jdk/incubator/code/type/JavaRef.java

```diff
@@ -0,0 +1,28 @@
+package jdk.incubator.code.type;
+
+/**
+ * A symbolic reference to a Java class member or type, commonly containing
+ * symbolic names together with {@link JavaType symbolic descriptions} of Java types.
+ * <p>
+ * A symbolic Java reference can be resolved to a corresponding instance of its
+ * reflected representation, much like the symbolic description of a Java type
+ * can be resolved to an instance of {@link java.lang.reflect.Type Type}.
+ */
+public sealed interface JavaRef
```

Comment:
Shouldn't `TypeVarRef` also extend from this?

Reply:
I always thought of it as just a java type element, correlated with `java.lang.reflect.TypeVariable` since that is what it resolves to.

Why thinking about this PR and more broadly I was pondering whether to rename `TypeVarRef` to `TypeVarRefType` or more verbosely as `TypeVariableType`, since i must admit to  to finding the `Ref` in `TypeVarRef` a little confusing as presumably it is referring to the type variable, but then `ClassType` refers to the class, etc.

Reply:
> I always thought of it as just a java type element, correlated with `java.lang.reflect.TypeVariable` since that is what it resolves to.
> 
> Why thinking about this PR and more broadly I was pondering whether to rename `TypeVarRef` to `TypeVarRefType` or more verbosely as `TypeVariableType`, since i must admit to to finding the `Ref` in `TypeVarRef` a little confusing as presumably it is referring to the type variable, but then `ClassType` refers to the class, etc.

Yeah - we went down the "ref" path because, well, it points to a declaration (of a type-variable). But I agree that `TypeVariable` or `TypeVariableType` is the better name. This is squarely a "type" - something that can be used where a type is expected, and something that can be turned into a `j.l.r.Type`